### PR TITLE
Create tests for `worker.py` file

### PR DESCRIPTION
**Pull request by yanazaeva**
|Related issue|
|---|
|#9999|

## Description

Hi team,

This PR was created in order to update the unit tests. Raised the test coverage up to 100% for the `test_worker.py` test.

## Tests results
```
Name                                     Stmts   Miss  Cover   Missing
----------------------------------------------------------------------
framework/wazuh/core/cluster/worker.py     370      0   100%
----------------------------------------------------------------------
TOTAL                                      370      0   100%
```

### CODE REVIEW

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 1]
```diff
@@ -21,223 +22,1068 @@
...
+            pass
+
+    # Test the first if
+    worker_handler.remove_bulk_agents(None, LoggerMock())
+
+    # Test the rest of the function
+
+    with patch.object(LoggerMock, "debug") as logger_debug_mock:
+        with patch.object(LoggerMock, "debug2") as logger_debug2_mock:
+            with patch("shutil.rmtree") as rmtree_mock:
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/tests/test_worker.py:839]: Remove all unused alias

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 2]
```diff
@@ -21,223 +22,1068 @@
...
+                worker_handler.remove_bulk_agents(["None"], LoggerMock())
+
+            isdir_mock.return_value = False
+            with patch("os.remove") as os_remove_mock:
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/tests/test_worker.py:843]: Remove all unused alias

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 3]
```diff
@@ -21,223 +22,1068 @@
...
+                worker_handler.remove_bulk_agents(["None"], LoggerMock())
+
+            logger_debug_mock.assert_has_calls(
+                [call("Removing files from 1 agents"), call("Agents to remove: None"),
+                 call("Agent files removed")])
+            logger_debug2_mock.assert_has_calls(
+                [call("Removing files from agents None"),
+                 call("Removing agent group assigments from database")], any_order=True)
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/tests/test_worker.py:851]: Typo here and inside the tested function

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 4]
```diff
@@ -246,11 +1092,9 @@ async def check_message(expected_messages, *args, **kwargs):
             for i, expected_message in enumerate(expected_messages):
                 assert expected_message in caplog.records[-(i+1)].message
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/tests/test_worker.py:1093]:
```suggestion
                assert expected_message in caplog.records[-(i + 1)].message

```

**Review by AdriiiPRodri — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 5]
```diff
@@ -2,15 +2,16 @@
...
-from unittest.mock import patch, mock_open, MagicMock, call
-
 import pytest
+import sys
+import threading
+import time
 import uvloop
-
-from wazuh.core.exception import WazuhException
+import os
```
[the remark this answers is not in the document]
Comment (yanazaeva) [reply framework/wazuh/core/cluster/tests/test_worker.py:12]: This one is used.

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 6]
```diff
@@ -2,15 +2,16 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 import asyncio
+import json
```
[the remark this answers is not in the document]
Comment (yanazaeva) [reply framework/wazuh/core/cluster/tests/test_worker.py:5]: This one is used too.

**Review by AdriiiPRodri — COMMENTED** [review]
This does not match with what is indicated in the issue (https://github.com/wazuh/wazuh/issues/9999#issuecomment-959405933): 
```
Name                                     Stmts   Miss  Cover   Missing
----------------------------------------------------------------------
framework/wazuh/core/cluster/worker.py     421     27    94%   791-802, 807-832, 872-875, 883-885
----------------------------------------------------------------------
TOTAL                                      421     27    94%
```

**Comment by yanazaeva** [comment]
The checks executed in this commit were constantly failing due to the error attached [here.](https://github.com/wazuh/wazuh/files/7925529/test_worker_failure.zip) 

We had run these tests in a docker image that contains the exact environment used by Jenkins, and the tests were randomly passing and failing, so this pass seems to be random too. Further investigation is needed.

**Review by AdriiiPRodri — APPROVED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**